db/db.go

```diff
@@ -2386,26 +2386,19 @@ func GetValidatorsWithdrawals(validators []uint64, fromEpoch uint64, toEpoch uin
 	return withdrawals, nil
 }
 
-func GetValidatorsWithdrawalsByEpoch(validator []uint64, limit uint64, offset uint64) ([]*types.WithdrawalsByEpoch, error) {
+func GetValidatorsWithdrawalsByEpoch(validator []uint64, startEpoch uint64, endEpoch uint64) ([]*types.WithdrawalsByEpoch, error) {
 	var withdrawals []*types.WithdrawalsByEpoch
-	if limit == 0 {
-		limit = 100
-	}
-
-	if limit > (offset + 100) {
-		limit = offset + 100
-	}
 
 	err := ReaderDb.Select(&withdrawals, `
 	SELECT 
 		w.validatorindex,
 		w.block_slot / $4 as epoch, 
 		sum(w.amount) as amount
 	FROM blocks_withdrawals w
-	INNER JOIN blocks b ON b.blockroot = w.block_root AND b.status = '1'
-	WHERE validatorindex = ANY($1)
+	INNER JOIN blocks b ON b.blockroot = w.block_root AND b.status = '1' AND b.slot >= $2 AND b.slot <= $3
+	WHERE validatorindex = ANY($1) 
 	GROUP BY w.validatorindex, w.block_slot / $4
-	ORDER BY w.block_slot / $4 DESC LIMIT $2 OFFSET $3`, pq.Array(validator), limit, offset, utils.Config.Chain.Config.SlotsPerEpoch)
+	ORDER BY w.block_slot / $4 DESC LIMIT 100`, pq.Array(validator), startEpoch*utils.Config.Chain.Config.SlotsPerEpoch, endEpoch*utils.Config.Chain.Config.SlotsPerEpoch, utils.Config.Chain.Config.SlotsPerEpoch)
```

Comment:
**Minor:** This is good, passing something like `startEpoch*utils.Config.Chain.Config.SlotsPerEpoch` and using it straight away to compare it with a slot should be faster than passing only the epoch and dividing the slot by `utils.Config.Chain.Config.SlotsPerEpoch` for every row within the Postgres query.

This could be done in `GetWithdrawals`, `GetValidatorsWithdrawals` and `GetBLSChanges` too (I have only checked this very file for that).

Reply:
This needs to evaluated on an individual basis and should be done in a separate PR

Reply:
BIDS-1675